### ISSUE TITLE
Method comments in generated C++ are misplaced

### ISSUE THREAD

**Issue by larsfillmore** [dtolnay/cxx#829]
Rust struct method comments are not placed before the corresponding generated C++ class methods:

`cxx = "1.0.43"`
```rust
// lib.rs

#[cxx::bridge]
mod ffi {
    unsafe extern "C++" {
        include!("rust/cxx.h");
    }

    extern "Rust" {
        /// This struct comment is placed correctly.
        type SomeType;

        /// This method comment is misplaced.
        fn misplaced_comment(self: &SomeType);

        /// This free function comment is placed correctly.
        fn correctly_placed_comment();
    }
}

struct SomeType {}

impl SomeType {
    fn misplaced_comment(&self) {}
}

fn correctly_placed_comment() {}

```

this results in

```c++
// lib.rs.h

#pragma once
#include "rust/cxx.h"

struct SomeType;

#ifndef CXXBRIDGE1_STRUCT_SomeType
#define CXXBRIDGE1_STRUCT_SomeType
// This struct comment is placed correctly.
struct SomeType final : public ::rust::Opaque {
  void misplaced_comment() const noexcept;
  ~SomeType() = delete;

private:
  friend ::rust::layout;
  struct layout {
    static ::std::size_t size() noexcept;
    static ::std::size_t align() noexcept;
  };
};
#endif // CXXBRIDGE1_STRUCT_SomeType

// This method comment is misplaced.

// This free function comment is placed correctly.
void correctly_placed_comment() noexcept;
```

**Comment by dtolnay** [comment]
Good catch -- I've published a fix in 1.0.44.

**Comment by larsfillmore** [comment]
Thanks for the quick fix!
